models/spring-ai-anthropic/src/main/java/org/springframework/ai/anthropic/api/AnthropicApi.java

```diff
@@ -217,7 +231,7 @@ public Flux<ChatCompletionResponse> chatCompletionStream(ChatCompletionRequest c
 		return this.webClient.post()
 			.uri(this.completionsPath)
 			.headers(headers -> {
-				headers.addAll(additionalHttpHeader);
+				headers.addAll(HttpHeaders.readOnlyHttpHeaders(additionalHttpHeader));
```

Comment:
This works however it is less performant as it keeps creating new objects even when `additionalHttpHeader` is empty. I would advise something more like:

```java
additionalHttpHeader.forEach(headers::addAll);
```